public/app/features/alerting/unified/components/rules/deleted-rules/DeletedRules.tsx

```diff
@@ -0,0 +1,131 @@
+import { useState } from 'react';
+
+import { dateTimeFormat, dateTimeFormatTimeAgo } from '@grafana/data';
+import { Button, Column, InteractiveTable, Stack } from '@grafana/ui';
+import { Trans, t } from 'app/core/internationalization';
+import { GrafanaRuleDefinition, RulerGrafanaRuleDTO } from 'app/types/unified-alerting-dto';
+
+import { trackDeletedRuleRestoreFail, trackDeletedRuleRestoreSuccess } from '../../../Analytics';
+import { UpdatedByUser } from '../../rule-viewer/tabs/version-history/UpdatedBy';
+
+import { ConfirmRestoreDeletedRuleModal } from './ConfirmRestoreDeletedRuleModal';
+
+export const DELETED_RULES_PAGE_SIZE = 30;
+
+interface DeletedRulesProps {
+  deletedRules: Array<RulerGrafanaRuleDTO<GrafanaRuleDefinition>>
+}
+export function DeletedRules({ deletedRules }: DeletedRulesProps) {
+  const [confirmRestore, setConfirmRestore] = useState(false);
+  const [restoreRule, setRestoreRule] = useState<RulerGrafanaRuleDTO | undefined>();
+
+  const unknown = t('alerting.deletedRules.unknown', 'Unknown');
+
+  const showConfirmation = (id: string) => {
+    const ruleTorestore = deletedRules.find(
+      (rule) => getRowId(rule.grafana_alert) === id
+    );
+    if (!ruleTorestore) {
+      return;
+    }
+
+    setConfirmRestore(true);
+    setRestoreRule(ruleTorestore);
+  };
+
+  const hideConfirmation = () => {
+    setConfirmRestore(false);
+  };
+
+  const columns: Array<Column<(typeof deletedRules)[0]>> = [
+    {
+      id: 'createdBy',
+      header: t('alerting.deletedRules.table.updatedBy', 'Deleted By'),
+      disableGrow: true,
+      cell: ({ row }) => {
+        return <UpdatedByUser user={row.original.grafana_alert.updated_by} />;
+      },
+    },
+    {
+      id: 'title',
+      header: t('alerting.deletedRules.table.title', 'Title'),
+      disableGrow: true,
+      cell: ({ row }) => {
+        return row.original.grafana_alert.title;
+      },
+    },
+    {
+      id: 'folder',
+      header: t('alerting.deletedRules.table.folder', 'Folder'),
+      disableGrow: true,
+      cell: ({ row }) => {
+        return row.original.grafana_alert.namespace_uid;
```

Comment:
Would be nice to show the namespace title rather than UID, but we'd have to fetch each from the API 🤔

Reply:
yes, exactly.... 🫠